source/MaterialXCore/Interface.cpp

```diff
@@ -64,13 +65,13 @@ bool PortElement::validate(string* message) const
             {
                 OutputPtr output = connectedNodeDef->getOutput(getOutputString());
                 validateRequire(output != nullptr, res, message, "Invalid output in port connection");
-                if (output)
+                if (output && !hasChannels())
```

Comment:
Should we check that at least the type is a vector/color or float. If it's another type channels is also not valid (e.g. matrix)

Reply:
Added a pretty extensive addition related to this.